include/triton/core/tritonserver.h

```diff
@@ -378,6 +378,31 @@ typedef TRITONSERVER_Error* (*TRITONSERVER_ResponseAllocatorReleaseFn_t)(
 typedef TRITONSERVER_Error* (*TRITONSERVER_ResponseAllocatorStartFn_t)(
     TRITONSERVER_ResponseAllocator* allocator, void* userp);
 
+/// Type for function that is called to query the allocator's preferred memory
+/// type and memory type ID, which will be the attributes of the allocated
+/// buffers if explicitly requested. 'tensor_name' and 'byte_size' are optional
+/// parameters for the allocator. Note that if 'tensor_name' or 'byte_size' is
+/// not provided, the returned values may not match the attributes of the
+/// allocated buffer even if explicitly requested, as the allocator may allocate
+/// buffers differently based on the requested paramters.
+///
+/// \param allocator The allocator that is provided in the call to
+/// TRITONSERVER_InferenceRequestSetResponseCallback.
+/// \param buffer_userp The user-specified value associated
+/// \param tensor_name The name of the output tensor. This is optional
+/// and it should be set to nullptr to indicate that the tensor name has
+/// not determined.
+/// \param byte_size The expected size of the buffer. This is optional
+/// and it should be set to nullptr to indicate that the byte size has
+/// not determined.
+/// \param memory_type The preferred type of memory.
```

Comment:
Make these in/out paramaters just like in alloc_fn. That way the backend can communicate its preference.